test/e2e/configuration_anomaly_detection_test.go

```diff
@@ -672,4 +668,135 @@ var _ = Describe("Configuration Anomaly Detection", Ordered, func() {
 		}
 	})
 
+	It("UpgradeConfigSyncFailureOver4Hr: corrupted pull secret investigation", Label("pull-secret", "upgrade-config-sync", "user-banned-check"), func(ctx context.Context) {
+		// Get cluster information from OCM
+		response, err := ocme2eCli.ClustersMgmt().V1().Clusters().Cluster(clusterID).Get().Send()
+		Expect(err).ToNot(HaveOccurred(), "failed to get cluster from OCM")
+		cluster := response.Body()
+		Expect(cluster).ToNot(BeNil(), "received nil cluster from OCM")
+
+		lsResponseBefore, err := utils.GetLimitedSupportReasons(ocme2eCli, clusterID)
+		var lsReasonsBefore int
+		if err != nil {
+			ginkgo.GinkgoWriter.Printf("Could not get limited support reasons before test: %v\n", err)
+			lsReasonsBefore = 0
+		} else {
+			lsReasonsBefore = lsResponseBefore.Items().Len()
+			ginkgo.GinkgoWriter.Printf("Limited support reasons before pull secret corruption/user banned check: %d\n", lsReasonsBefore)
+		}
+
+		// Check if user is banned (part of the investigation logic)
+		ginkgo.GinkgoWriter.Printf("Checking if cluster owner is banned...\n")
+		userBannedStatus, userBannedNotes, err := utils.IsUserBanned(ocme2eCli, cluster)
+		Expect(err).NotTo(HaveOccurred(), "Failed to check if user is banned")
+
+		if userBannedStatus {
+			ginkgo.GinkgoWriter.Printf("User is banned: %s\n", userBannedNotes)
+			_, err = testPdClient.TriggerIncident("UpgradeConfigSyncFailureOver4HrSRE", clusterID)
```

Comment:
If the user is banned this triggers and incident and later on triggers the same incident again.